java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -1915,6 +1915,96 @@ void testCastNegativeTimestampAsString() {
+    List<Byte> list4 = Arrays.asList((byte)0x41, (byte)0x52, (byte)0xc3, (byte)0xa9);
+    List<Byte> list5 = Arrays.asList((byte)0x5c, (byte)0x54, (byte)0x48, (byte)0x45, (byte)0x09, (byte)0x38, (byte)0xed, (byte)0x9c, (byte)0xa0);
+    List<Byte> list6 = Arrays.asList((byte)0x74, (byte)0xc3, (byte)0xa9, (byte)0x73, (byte)0x74, (byte)0x20, (byte)0x73, (byte)0x74, (byte)0x72, (byte)0x69, (byte)0x6e, (byte)0x67, (byte)0x73);
+    List<Byte> list7 = Arrays.asList();
+    List<Byte> list8 = Arrays.asList((byte)0xc3, (byte)0xa9, (byte)0xc3, (byte)0xa9);
+
+    try(ColumnVector cv = ColumnVector.fromStrings("Thésé", null, "\r\ud720é\ud721", "ARé",
+    "\\THE\t8\ud720", "tést strings", "", "éé");
+        ColumnVector res = cv.asByteList(true);
+        HostColumnVector host_res = res.copyToHost()) {
+        List<Byte> ret1 = host_res.getList(0);
+        List<Byte> ret2 = host_res.getList(1);
+        List<Byte> ret3 = host_res.getList(2);
+        List<Byte> ret4 = host_res.getList(3);
+        List<Byte> ret5 = host_res.getList(4);
+        List<Byte> ret6 = host_res.getList(5);
+        List<Byte> ret7 = host_res.getList(6);
+        List<Byte> ret8 = host_res.getList(7);
+        assertEquals(list1, ret1, "Lists don't match");
+        assertEquals(list2, ret2, "Lists don't match");
+        assertEquals(list3, ret3, "Lists don't match");
+        assertEquals(list4, ret4, "Lists don't match");
+        assertEquals(list5, ret5, "Lists don't match");
+        assertEquals(list6, ret6, "Lists don't match");
+        assertEquals(list7, ret7, "Lists don't match");
+        assertEquals(list8, ret8, "Lists don't match");
+    }
+  }
+
+  @Test
+  void testCastIntegerToByteList() {
+    List<Byte> list1 = Arrays.asList((byte)0x00, (byte)0x00, (byte)0x00, (byte)0x00);
+    List<Byte> list2 = Arrays.asList((byte)0x00, (byte)0x00, (byte)0x00, (byte)0x64);
+    List<Byte> list3 = Arrays.asList((byte)0xff, (byte)0xff, (byte)0xff, (byte)0x9c);
+    List<Byte> list5 = Arrays.asList((byte)0x80, (byte)0x00, (byte)0x00, (byte)0x00);
+    List<Byte> list6 = Arrays.asList((byte)0x7f, (byte)0xff, (byte)0xff, (byte)0xff);
+
+    try(ColumnVector cv = ColumnVector.fromBoxedInts(0, 100, -100, null, Integer.MIN_VALUE, Integer.MAX_VALUE);
+        ColumnVector res = cv.asByteList(true);
+        HostColumnVector host_res = res.copyToHost()) {
+        List<Byte> ret1 = host_res.getList(0);
+        List<Byte> ret2 = host_res.getList(1);
+        List<Byte> ret3 = host_res.getList(2);
+        boolean ret4 = host_res.isNull(3);
+        List<Byte> ret5 = host_res.getList(4);
+        List<Byte> ret6 = host_res.getList(5);
+        assertEquals(list1, ret1, "Lists don't match");
+        assertEquals(list2, ret2, "Lists don't match");
+        assertEquals(list3, ret3, "Lists don't match");
+        assertTrue(ret4, "Lists don't match");
+        assertEquals(list5, ret5, "Lists don't match");
+        assertEquals(list6, ret6, "Lists don't match");
```

Comment:
Can use assertColumnsAreEqual as well - nit.

Reply:
Using `assertColumnsAreEqual` fails with the following error.

```
[ERROR]     method ai.rapids.cudf.TableTest.assertColumnsAreEqual(ai.rapids.cudf.HostColumnVector,ai.rapids.cudf.HostColumnVector,java.lang.String) is not applicable
[ERROR]       (argument mismatch; java.util.List<java.lang.Byte> cannot be converted to ai.rapids.cudf.HostColumnVector)
[ERROR]     method ai.rapids.cudf.TableTest.assertColumnsAreEqual(ai.rapids.cudf.ColumnVector,ai.rapids.cudf.ColumnVector,java.lang.String) is not applicable
[ERROR]       (argument mismatch; java.util.List<java.lang.Byte> cannot be converted to ai.rapids.cudf.ColumnVector)
```

Reply:
I think what @kuhushukla wanted was to create an expected ColumnVector out of `list*` and then call assertColumnsAreEqual

Reply:
That is right.

Reply:
`ColumnVector expected_int8 = ColumnVector.fromLists(new HostColumnVector.ListType(true, new HostColumnVector.BasicType(true, DType.UINT8)), list1, list2, list3, list4, list5, list6, list7);` causes 
```
java.lang.AssertionError
        at ai.rapids.cudf.HostColumnVector$ColumnBuilder.append(HostColumnVector.java:871)
        at ai.rapids.cudf.HostColumnVector$ColumnBuilder.appendChildOrNull(HostColumnVector.java:848)
        at ai.rapids.cudf.HostColumnVector$ColumnBuilder.append(HostColumnVector.java:822)
        at ai.rapids.cudf.HostColumnVector$ColumnBuilder.appendLists(HostColumnVector.java:733)
        at ai.rapids.cudf.HostColumnVector.fromLists(HostColumnVector.java:277)
        at ai.rapids.cudf.ColumnVector.fromLists(ColumnVector.java:3583)
```
It has something to do with `DType.UINT8` and the `fromLists` function. Since `castTo` does not support list types and `fromLists` warns about potential issues, so I'm going to keep them as is for now.